docs/source/conf-schemas-examples.rst

```diff
@@ -0,0 +1,384 @@
+        "parent": {
+          "type": "string",
+          "name": "string",
+          "id": "integer"
+        }
+      },
+      "created_by": {
+        "type": "string",
+        "id": "integer",
+        "name": "string",
+        "login": "string"
+      },
+      "created_at": "string",
+      "event_id": "string",
+      "event_type": "string",
+      "ip_address": "string",
+      "type": "string",
+      "session_id": "string",
+      "additional_details": {}
+    },
+    "parser": "json"
+  },
+
+Cloudwatch VPC Flow Logs
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+AWS VPC Flow Logs can be delivered to StreamAlert via CloudWatch.
+
+As they are compressed with deflate, we can use the special ``gzip-json`` for parsing and analysis.
+
+CloudWatch logs are delivered as a nested record, so we will need to pass ``configuration`` options to the parser to find the nested records::
+
+  "cloudwatch:flow_logs": {
+    "schema": {
+      "protocol": "integer",
+      "source": "string",
+      "destination": "string",
+      "srcport": "integer",
+      "destport": "integer",
+      "action": "string",
+      "packets": "integer",
+      "bytes": "integer",
+      "windowstart": "integer",
+      "windowend": "integer",
+      "version": "integer",
+      "eni": "string",
+      "account": "integer",
+      "flowlogstatus": "string"
+    },
+    "parser": "gzip-json",
+    "configuration": {
+      "json_path": "logEvents[*].extractedFields",
+      "envelope_key": {
+        "logGroup": "string",
+        "logStream": "string",
+        "owner": "integer"
+      }
+    }
+  }
+
+osquery
+~~~~~~~
+
+osquery's schema depends on the table being queried and there are 50+ tables
+
+**Option 1**: Define a schema for each table used::
+
+  "osquery:etc_hosts": {
+    "parser": "json",
+    "schema": {
+      "name": "string",
+      ...
+      "columns": {
+        "address": "string",
+        "hostnames": "string"
+      },
+      "action": "string",
+      ...
+    }
+  },
+  "osquery:listening_ports": {
+    "parser": "json",
+    "schema": {
+      "name": "string",
+      ...
+      "columns": {
+        "pid": "integer",
+        "port": "integer",
+        "protocol": "integer",
+        ...
+      },
+      "action": "string",
+      ...
+    }
+  },
+  ...
+
+This promotes Rule safety, but requires additional time to define the schemas
+
+
+**Option 2**: Define a "loose" schema that captures all tables::
+
+  "osquery": {
+    "parser": "json",
+    "schema": {
+      "name": "string",
+      "hostIdentifier": "string",
+      "calendarTime": "string",
+      "unixTime": "integer",
+      "columns": {},                 # {} = any keys
+      "action": "string"
+    }
+  },
+
+.. warning:: In Option 2, the schema definition is flexible, but Rule safety is lost because you'll need to use defensive programming when accessing and analyzing fields in `columns`. The use of `req_subkeys` will be advised, see Rules for more details
```

Comment:
The `warning` is not rendering in GitHub as anything special. Is it supposed to be?